File: src/editor/render/colors.ts
import { isBetween } from './helpers';
import { ColorSaturationData } from './index';

export function isRed(data: ColorSaturationData): boolean {
    return data.hue > 340 || data.hue <= 10;
}

export function isOrange(data: ColorSaturationData): boolean {
    return isBetween(data.hue, 10, 40);
}

export function isYellow(data: ColorSaturationData): boolean {
    return isBetween(data.hue, 40, 60);
}

export function isGreen(data: ColorSaturationData): boolean {
    return isBetween(data.hue, 60, 130);
}
export function isTeal(data: ColorSaturationData): boolean {
    return isBetween(data.hue, 130, 160);
}

export function isCyan(data: ColorSaturationData): boolean {
    return isBetween(data.hue, 160, 220);
}

export function isBlue(data: ColorSaturationData): boolean {
    return isBetween(data.hue, 220, 260);
}

export function isPurple(data: ColorSaturationData): boolean {
    return isBetween(data.hue, 260, 300);
}

export function isMagenta(data: ColorSaturationData): boolean {
    return isBetween(data.hue, 300, 340);
}

File: src/editor/renderEngine/colors.c
#include <stdbool.h>
#include "helpers.h"
#include <emscripten/emscripten.h>

bool isRed(double hue) {
    if (hue > 340 || hue <= 10) {
        return 1;
    }
    return 0;
}

bool isOrange(double hue) {
    return isBetween(hue, 10, 40);
}
bool isYellow(double hue) {
    return isBetween(hue, 40, 60);
}
bool isGreen(double hue) {
    return isBetween(hue, 60, 130);
}
bool isTeal(double hue) {
    return isBetween(hue, 130, 160);
}
bool isCyan(double hue) {
    return isBetween(hue, 160, 220);
}
bool isBlue(double hue) {
    return isBetween(hue, 220, 260);
}
bool isPurple(double hue) {
    return isBetween(hue, 260, 300);
}
bool isMagenta(double hue) {
    return isBetween(hue, 300, 340);
}